chemdash.py

```diff
@@ -368,13 +354,45 @@ def get_dist_plot(selected_columns, data):
 
         return select_cols, fig
 
+    return app
+
+
+def main(
+    input_file: Path,
+    port: int = 8000,
+    debug: bool = False
+):
+    """
+    Run the chemdash server.
 
-else:
-    print ('Your dataset needs to contain "smiles" and "Compound_id" columns and a maximum of 10k cpds')
+    :param input_file: Path to the input file of compounds.
+    :param port: Port number to use (default is 8000).
+    :param debug: Debug mode (default is False).
+    """
+    df = pd.read_csv(str(input_file))
+    if not all(x in df.columns for x in ['smiles', 'Compound_id']):
+        raise ValueError('Your dataset needs to contain "smiles" and "Compound_id" columns!')
+    if df.shape[0] > MAX_COMPOUNDS:
+        raise ValueError('This tool supports a maxiumum of 10k compounds!')
+
+    typer.echo('There are ' + str(df.shape[0]) + ' compounds in your dataset')
+
+    # create asset directory for mol image files
+    if not os.path.isdir("assets/mol_images"):
+        os.mkdir('assets/mol_images')
+
+    df = prepare_data(df)
+    df['id'] = df.index
+
+    app = create_chemdash_app(df)
+
+    typer.echo(f"Launching server: http://0.0.0.0:{port}, debug: {debug}")
+    app.run_server(host="0.0.0.0", port=port, debug=debug)
+    typer.echo(f"Server exited.")
 
 
 if __name__ == '__main__':
-    port = 8000
-    print(f"Launching server: http://0.0.0.0:{port}")
-    app.run_server(host="0.0.0.0", port=port, debug=True)
-    print(f"Server exited.")
+    # NOTE: THe parallelization logic in prepare() will re-load this module in sub-processes,
+    # All "real work" is done in this block, with the main module body doing only definitions!
+    typer.run(main)
```

Comment:
We now have just the typer.run() call in `__name__ == "__main__"`.

Reply:
love this!